hub.go

```diff
@@ -488,6 +488,14 @@ func (h *Hub) GetRuntimeInformation(ctx context.Context) (*HubRuntimeInformation
 	defer span.End()
 	client := newClient(h.namespace, h.name)
 	c, err := h.namespace.newConnection()
+	if c != nil {
```

Comment:
Is it possible for `c` to be valid when `err != nil`? In common usage it's assumed that the first value shouldn't be used in any way if `err != nil` and a defer would typically be placed after the error check. (Unless documented otherwise.)

Reply:
Awesome question. I wasn't sure if c.start() should return something that raises an error, but still needed to be closed. Having now dug deeper, it appears that if an error is returned below, you are doing the right thing by closing c.

Thank you for the catch. I will make the change.